include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp

```diff
@@ -254,6 +254,34 @@ class two_dimensional_matrix
     }
     //!\}
 
+    /*!\brief Explicit conversion to other matrix major order.
```

Comment:
I only understood this (fully) after looking the tests.
So this is kinda like transposing without changing the dimensions.

Maybe you can add a sentence or two in the details and/or add the test example:

```
Before:
0  1  2  3
4  5  6  7
8  9  10 11

After:
0  3  6  9
1  4  7  10
2  5  8  11
```